core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -347,10 +347,10 @@ private[spark] class Worker(
             }.toSeq
           }
           appDirectories(appId) = appLocalDirs
-
-          val manager = new ExecutorRunner(appId, execId, appDesc, cores_, memory_,
-            self, workerId, host, sparkHome, executorDir, akkaUrl, conf, appLocalDirs,
-            ExecutorState.LOADING)
+          val manager = new ExecutorRunner(appId, execId,
+            appDesc.copy(command = Worker.maybeUpdateSSLSettings(appDesc.command, conf)),
```

Comment:
Hmmm... why do you need the copy? A quick overlook of `ExecutorRunner` doesn't seem to indicate it modifies this object in any way...